internal/services/watchlist/watchlist.go

```diff
@@ -0,0 +1,62 @@
+package watchlist
+
+import (
+	"Tradeasy/config"
+	"Tradeasy/internal/model"
+	"errors"
+)
+
+func CreateWatchlist(CrReq CreateRequest) (CrRes CreateResponse, err error) {
+	var wl model.Watchlist
+	wl.Name = CrReq.WatchlistName
+	er := config.DB.Table("watchlist").Where("name = ?", CrReq.WatchlistName).First(&wl).Error
+	if er == nil {
+		return CrRes, errors.New("watchlist name already exists")
+	}
+	err = config.DB.Table("watchlist").Create(&wl).Error
+	if err != nil {
+		return CrRes, errors.New("failed to create watchlist")
+	}
+	CrRes.Message = "Watchlist created"
+	CrRes.WatchlistId = wl.Id
+
+	return CrRes, nil
+}
+
+func AddStockEntry(AddReq AddStockRequest, watchlistId int) (AddRes AddStockResponse, err error) {
+	var uwl model.UserWatchlist
+	uwl.Userid = AddReq.UserId
+	uwl.WatchlistId = watchlistId
+	uwl.StockName = AddReq.StockName
+
+	er := config.DB.Table("user_watchlist").Where("stock_name = ? AND watchlist_id = ? AND user_id = ?", AddReq.StockName, watchlistId, AddReq.UserId).First(&uwl).Error
+	if er == nil {
+		return AddRes, errors.New("stock name already exists")
+	}
+	err = config.DB.Table("user_watchlist").Create(&uwl).Error
+	if err != nil {
+		return AddRes, errors.New("stock not added")
+	}
+	AddRes.Message = "Stock added"
+	return AddRes, nil
+}
+
+func DeleteStockEntry(DelReq DeleteStockRequest, watchlistId int) (DelRes DeleteStockResponse, err error) {
+	var uwl model.UserWatchlist
+	err = config.DB.Table("user_watchlist").Where("user_id = ? AND watchlist_id = ? AND stock_name = ?", DelReq.UserId, watchlistId, DelReq.StockName).Delete(&uwl).Error
```

Comment:
Check if stock name exists for that watchlist and user or not.

Reply:
done